### PR TITLE
task docs improvements

### PR DESCRIPTION
**Pull request by casperdcl**
- add links to authentication credentials docs
- explain "auto-destroy unused resources"
- explain `script` usage (heredoc, `file()`, relative to `workdir.input`)

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -38,19 +38,25 @@ $ terraform apply
...
+See the [Azure documentation](https://docs.microsoft.com/en-us/python/api/azure-identity/azure.identity.environmentcredential) for more information.
+
 ### Google Cloud Platform
 
-- `GOOGLE_APPLICATION_CREDENTIALS` - Path to a service account JSON key file.
+- `GOOGLE_APPLICATION_CREDENTIALS` - Path to a [service account](https://cloud.google.com/iam/docs/service-accounts) JSON key file.
 
 -> **Note:** you can also use `GOOGLE_APPLICATION_CREDENTIALS_DATA` with the **contents** of the service account JSON key file.
 
+See the [summary of GCP credentials on the DVC site](https://dvc.org/doc/user-guide/setup-google-drive-remote#using-service-accounts) to get started quickly.
```
Comment (0x2b3bfa0) [inline docs/index.md:58]: 🙀 That guide is tailored for Google Drive, and there are some mislaeding details, like the `GDRIVE_CREDENTIALS_DATA` environment variable. Can't we just link to the official documentation?

```suggestion
See [this guide](https://cloud.google.com/docs/authentication/getting-started) to get started quickly.
```

**Review by 0x2b3bfa0 — CHANGES_REQUESTED** [review]
💅🏼 Added some nitpicks.

**Comment by DavidGOrtega** [comment]
Can we add also the spot instance lifecycle?

File: docs/index.md (hunk 1 above)
Comment (casperdcl) [reply docs/index.md:58]: Hmm good point. Should we also mention https://cloud.google.com/iam/docs/service-accounts? Perhaps not needed.

File: README.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ The Iterative Provider makes it easy to:
 - Rapidly move local machine learning experiments to a cloud infrastructure
 - Take advantage of training models on spot instances without losing any progress
 - Unify configuration of various cloud compute providers
-- Automatically destroy unused cloud resources (never forget to turn your GPU off again)
+- Automatically destroy unused cloud resources (compute instances are terminated on job completion/failure, and storage is removed when results are downloaded)
```
[the remark this answers is not in the document]
Comment (casperdcl) [reply README.md:13]: - "resource" means compute & bucket, while "instance" means only compute to me.
- "terminated" also seems more semantically correct for computer to me.

[the remark this answers is not in the document]
Comment (0x2b3bfa0) [reply README.md:13]: > "resource" means compute & bucket

Good point, although “compute resource” clarifies the ambiguity.

> while "instance" means only compute to me.

To my mind “instance” is an overloaded and unintuitive term that happens to be widely understood by cloud users.

Cloud [virtual] machine might be a more generic and accurate term, but it's less popular than “instance” on clouds other than Microsoft Azure.

[the remark this answers is not in the document]
Comment (0x2b3bfa0) [reply README.md:13]: “Terminated” is similar to “instance” well understood in this context but probably not as easy to understand as “deleted” or “removed”

[the remark this answers is not in the document]
Comment (0x2b3bfa0) [reply README.md:13]: Anyway, just bikeshedding. It's fine as is.

File: docs/index.md (hunk 1 above)
Comment (0x2b3bfa0) [reply docs/index.md:58]: That's the first thing they mention in the tutorial, so I guess that it's not worth a second link.

File: docs/resources/task.md [hunk 3]
```diff
@@ -30,7 +30,7 @@ resource "iterative_task" "task" {
 ### Required
 
 - `cloud` - (Required) Cloud provider to run the task on; valid values are `aws`, `gcp`, `az` and `k8s`.
-- `script` - (Required) Script to run; must begin with a valid [shebang](<https://en.wikipedia.org/wiki/Shebang_(Unix)>).
+- `script` - (Required) Script to run (relative to `workdir.input`); must begin with a valid [shebang](<https://en.wikipedia.org/wiki/Shebang_(Unix)>). Can use a string, including a [heredoc](https://www.terraform.io/docs/language/expressions/strings.html#heredoc-strings), or the contents of a file returned by the [`file`](https://www.terraform.io/docs/language/functions/file.html) function.
```
[the remark this answers is not in the document]
Comment (casperdcl) [reply docs/resources/task.md:33]: reworded the other description. Trying to keep it briefer (command ref style) here.

**Comment by casperdcl** [comment]
> Can we add also the spot instance lifecycle?

Already in #363 (as per https://github.com/iterative/terraform-provider-iterative/issues/397#issuecomment-1045053435)

**Review by 0x2b3bfa0 — APPROVED** [review]
🙏🏼 :shipit:

**Pull request merged**